salt/client/ssh/__init__.py

```diff
@@ -1029,13 +1034,24 @@ def check_thin_dir(self):
             return False
         return True
 
+    def check_venv_hash_file(self):
+        """
+        check if the venv exists on the remote machine
+        """
+        stdout, stderr, retcode = self.shell.exec_cmd(
+            "test -f {}".format(self.venv_hash_file)
+        )
+        return retcode == 0
+
     def deploy(self):
         """
         Deploy salt-thin
         """
-        self.shell.send(
-            self.thin, os.path.join(self.thin_dir, "salt-thin.tgz"),
-        )
+        if not self.check_venv_hash_file():
+            self.shell.send(
+                self.thin,
+                os.path.join(self.thin_dir, "salt-thin.tgz"),
+            )
```

Comment:
Does the check have to happen inside `deploy()` or can we just not call `deploy()` for systems that get `venv-salt-minion`? I'm not too deep into the salt-thin deployment code right now.

Reply:
Actually both questions are interconnected. I mean this one and the following. It depends on the result of running the pre flight script. Initially I tested it with returning success in all of the cases, but as we decided to support only Salt Bundle method of handling the events on the minion, the pre flight script will fail if it's not possible to deploy the Bundle. So it's safe to use this logic and just adjust the pre flight script to our needs. The pre flight script is not a part of any PR yet. Will be included in the separate PR after merging https://github.com/uyuni-project/uyuni/pull/4457.
But this PR is required to be merged first as https://github.com/uyuni-project/uyuni/pull/4457 can't work without this one.

Reply:
Ouch, sorry for the confusion https://github.com/uyuni-project/uyuni/pull/4457 requires not this one but https://github.com/openSUSE/salt/pull/475